proxy/object.go

```
@@ -0,0 +1,389 @@
+		if wr.Error != nil {
+			return wr.Error
+		}
+
+		// TODO : handle this error
+		// who closes the channel 'ch'?
+		stream.Send(&pb.WalkReply{
+			Key:           wr.Key,
+			Meta:          storMetaToPbMeta(wr.Meta),
+			Value:         wr.Data,
+			ReferenceList: wr.RefList,
+		})
+	}
+	return nil
+}
+
+func (osrv *objectSrv) AppendReferenceList(ctx context.Context, req *pb.ReferenceListRequest) (*pb.NullReply, error) {
+	err := checkReferenceListReq(req)
+	if err != nil {
+		return nil, err
+	}
+
+	if req.Meta != nil {
+		err = osrv.client.AppendReferenceListWithMeta(pbMetaToStorMeta(req.Meta), req.ReferenceList)
+	} else {
+		err = osrv.client.AppendReferenceList(req.Key, req.ReferenceList)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.NullReply{}, nil
+}
+
+func (osrv *objectSrv) RemoveReferenceList(ctx context.Context, req *pb.ReferenceListRequest) (*pb.NullReply, error) {
+	err := checkReferenceListReq(req)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Infof("ref list = %v", req.ReferenceList)
+
+	if req.Meta != nil {
+		err = osrv.client.RemoveReferenceListWithMeta(pbMetaToStorMeta(req.Meta), req.ReferenceList)
+	} else {
+		err = osrv.client.RemoveReferenceList(req.Key, req.ReferenceList)
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.NullReply{}, nil
+}
+
+func (osrv *objectSrv) Check(ctx context.Context, req *pb.CheckRequest) (*pb.CheckReply, error) {
+	if req.Key == nil {
+		return nil, ErrNilKey
+	}
+
+	status, err := osrv.client.Check(req.Key)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pb.CheckReply{
+		Status: storCheckStatusToPb(status),
+	}, nil
+}
+
+func (osrv *objectSrv) Repair(ctx context.Context, req *pb.RepairRequest) (*pb.NullReply, error) {
+	if req.Key == nil {
+		return nil, ErrNilKey
+	}
+
+	if err := osrv.client.Repair(req.Key); err != nil {
+		return nil, err
+	}
+
+	return &pb.NullReply{}, nil
+}
+
+func storCheckStatusToPb(cs client.CheckStatus) pb.CheckReply_Status {
+	switch cs {
```

you could use an internal `map[client.CheckStatus]pb.CheckReply_Status`.

Perhaps you can return an error so that this function returns `(pb.CheckReply_Status, error)`, might be better than to panic, as we never know what happens on the wire.